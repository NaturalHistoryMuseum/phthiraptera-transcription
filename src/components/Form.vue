<template>
  <form class="Form" method="POST">
    <input type="hidden" name="barcode" :value="barcode">
    <label class="Form__label">
      Locality
      <input class="Form__input">
    </label>
    <label class="Form__label">
      Host
      <input class="Form__input">
    </label>
    <label class="Form__label">
      Collection Date
      <input class="Form__input">
    </label>
    <label class="Form__label">
      Collector
      <input class="Form__input">
    </label>
    <label class="Form__label">
      Type Status
      <input class="Form__input">
    </label>
    <label class="Form__label">
      Registration Number
      <input class="Form__input">
    </label>
    <label class="Form__label">
      Sex/Stage
      <input class="Form__input">
    </label>

    <button>Submit</button>
  </form>
</template>

<script>
export default {
  props: ['barcode']
}
</script>

<style>
.Form {
  columns: 2 250px;
}

.Form__label {
  display: block;
  font-weight: bold;
  font-size: 0.9em;
  -webkit-column-break-inside: avoid; /* Chrome, Safari, Opera */
          page-break-inside: avoid; /* Firefox */
               break-inside: avoid;
}

.Form__input {
  display: block;
  margin: 3px 0 10px;
  padding: 5px;
  width: 100%;
}
</style>
