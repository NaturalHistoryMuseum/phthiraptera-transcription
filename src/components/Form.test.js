import Form from './Form.vue';
import { shallowMount } from '@vue/test-utils'

const propsData = {
	token: 'abc',
	scientificName: 'Sci name',
	suggestions: {},
	values: {},
	barcodes: [],
	thumbnails: [],
	action: ''
}

globalThis.fetch = () => new Promise(()=>{});

test('Form renders', () => {
	expect(shallowMount(Form, { propsData, provide: { eventBus: {} } }).html()).toMatchSnapshot();
});
