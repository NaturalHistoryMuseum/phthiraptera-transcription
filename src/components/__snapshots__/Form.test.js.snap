// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders 1`] = `
"<form method=\\"POST\\" action=\\"\\" class=\\"Form\\"> <input type=\\"hidden\\" name=\\"sciName\\" value=\\"Sci name\\">
  <dialog-stub></dialog-stub>
  <div class=\\"Form__wrapper\\"><input type=\\"hidden\\" name=\\"token\\" value=\\"abc\\">
    <div class=\\"Form__columns\\">
      <fieldset class=\\"Form__fieldset\\">
        <legend>Host</legend>
        <field-stub class=\\"Form__float-reset\\"></field-stub>
        <select-stub label=\\"Host type\\" name=\\"host_type\\">
          <option value=\\"No host\\">No host</option>
          <optgroup label=\\"Host present\\">
            <option value=\\"\\">Host present</option>
            <option value=\\"Skin\\">Skin</option>
            <option value=\\"Straggler / questionable host\\">Straggler / questionable host</option>
            <option value=\\"Other (nest, clothing etc)\\">Other (nest, clothing etc)</option>
          </optgroup>
        </select-stub>
      </fieldset>
      <fieldset class=\\"Form__fieldset\\">
        <legend>Locality</legend>
        <field-stub empty=\\"Real\\"></field-stub>
        <select-stub label=\\"Country\\" name=\\"country\\">
          <option selected=\\"selected\\"></option>
        </select-stub>
        <suggest-stub name=\\"precise_locality\\" label=\\"Precise locality\\"></suggest-stub>
      </fieldset>
      <fieldset class=\\"Form__fieldset\\">
        <legend>Collector(s)</legend>
        <field-stub value=\\"[object Object]\\" class=\\"Form__float-reset\\"></field-stub>
        <suggest-stub name=\\"collection\\" label=\\"Institution or collection\\"></suggest-stub>
      </fieldset>
      <fieldset class=\\"Form__fieldset\\">
        <legend>Collection Date</legend>
        <field-stub value=\\"[object Object]\\"></field-stub>
      </fieldset>
      <fieldset class=\\"Form__fieldset\\">
        <legend>Registration Number</legend>
        <input-stub name=\\"registration_number\\">
          <div class=\\"Form__label\\"><label for=\\"registration_number\\">
              BM number
            </label>
            <tooltip-stub>e.g. <em>BM 1980-1</em></tooltip-stub>
          </div>
        </input-stub>
      </fieldset>
      <fieldset class=\\"Form__fieldset\\">
        <legend>Type Status</legend>
        <field-stub empty=\\"\\"></field-stub>
      </fieldset>
      <fieldset class=\\"Form__fieldset\\">
        <legend>Sex/Stage</legend>
        <input-stub label=\\"Total number of specimens\\" name=\\"total_count\\" type=\\"number\\"></input-stub>
        <field-stub empty=\\"\\" value=\\"\\"></field-stub>
      </fieldset>
      <fieldset class=\\"Form__fieldset\\">
        <legend>Other</legend>
        <field-stub></field-stub>
        <field-stub></field-stub>
      </fieldset>
      <!---->
    </div>
  </div>
  <div class=\\"Form__footer\\">
    <div class=\\"Form__controls\\"><button class=\\"Form__submit Form__button\\">Submit</button></div>
  </div>
</form>"
`;
